src/engine.ml

```diff
@@ -861,40 +865,40 @@ let expected_packet session transport data =
      both from their side, and acks from our side *)
   let* () =
     guard
-      (Int32.equal session.their_packet_id hdr.Packet.packet_id)
-      (`Non_monotonic_packet_id (transport, hdr))
+      (Int32.equal session.their_replay_id hdr.Packet.replay_id)
+      (`Non_monotonic_replay_id (transport, hdr))
   in
   let+ () =
     opt_guard
-      (Int32.equal transport.their_message_id)
+      (Int32.equal transport.their_sequence_number)
       msg_id
-      (`Non_monotonic_message_id (transport, msg_id, hdr))
+      (`Non_monotonic_sequence_number (transport, msg_id, hdr))
```

Comment:
We should rename msg_id with `sequence_number`, and rename `Packet.message_id` as well.